parcels/kernel.py

```diff
@@ -250,11 +254,18 @@ def execute_jit(self, pset, endtime, dt):
 
         fargs = [byref(f.ctypes_struct) for f in self.field_args.values()]
         fargs += [c_double(f) for f in self.const_args.values()]
-        particle_data = pset._particle_data.ctypes.data_as(c_void_p)
-        self._function(c_int(len(pset)), particle_data,
-                       c_double(endtime),
-                       c_double(dt),
-                       *fargs)
+
+        for subfield_c in pset.particles_c:
```

Comment:
It's a bit confusing you call this `subfield`, because a `Field` is a specific object in Parcels, and I don't think that these subfields have anything to do with `Fields`